File: edumanage/springboot/src/main/java/com/keshe/edumanage/entity/base/TeachingTask.java
package com.keshe.edumanage.entity.base;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableName;
import com.keshe.edumanage.entity.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 教学任务（任课关系 + 上课时间）
 * <p>一条记录表示「某教师在某学期教某班级的某门课程，在星期几第几节、哪个教室上课」，
 * 是师生端「我的课程 / 我的教学任务 / 课表」与教师成绩录入权限的判断依据。</p>
 */
@Data
@TableName("base_teaching_task")
@EqualsAndHashCode(callSuper = true)
public class TeachingTask extends BaseEntity {
    /**
     * 教师ID
     */
    private Long teacherId;

    /**
     * 课程ID
     */
    private Long courseId;

    /**
     * 班级ID
     */
    private Long classId;

    /**
     * 学期ID
     */
    private Long termId;

    /**
     * 星期几（1=周一 … 7=周日），为空表示尚未排课
     */
    private Integer weekday;

    /**
     * 开始节次
     */
    private Integer startSection;

    /**
     * 结束节次
     */
    private Integer endSection;

    /**
     * 上课教室ID
     */
    private Long classroomId;

    /**
     * 上课周次，如 1-16周
     */
    private String weeks;

    /**
     * 以下为展示用字段（不属于 base_teaching_task 表），由 Service 关联填充。
     */
    @TableField(exist = false)
    private String courseName;

    @TableField(exist = false)
    private String className;

    @TableField(exist = false)
    private String teacherName;

    @TableField(exist = false)
    private String roomName;
}

File: student-teacher/src/api/teachingTask.ts
import { http, normalizeList } from '@/utils/request'

/** 教学任务（base_teaching_task；含上课时间，用于课表） */
export interface TeachingTask {
  id?: number
  teacherId?: number
  courseId: number
  classId: number
  termId?: number
  /** 星期几（1=周一 … 7=周日）；为空表示未排课 */
  weekday?: number
  startSection?: number
  endSection?: number
  classroomId?: number
  weeks?: string
  /** 展示用扩展字段（后端 Service 关联填充） */
  courseName?: string
  className?: string
  teacherName?: string
  roomName?: string
}

/** 星期几文案 */
export const WEEKDAY_TEXT: Record<number, string> = {
  1: '周一',
  2: '周二',
  3: '周三',
  4: '周四',
  5: '周五',
  6: '周六',
  7: '周日',
}

/**
 * 学生：我的课表（按班级查教学任务）
 * 后端 GET /api/teaching-tasks?classId=&termId=
 */
export async function listTasksByClass(classId: number, termId?: number): Promise<TeachingTask[]> {
  const data = await http.get<unknown>('/teaching-tasks', {
    classId,
    ...(termId ? { termId } : {}),
  })
  return normalizeList<TeachingTask>(data)
}

/**
 * 教师：我的教学任务（含上课时间）
 * 后端 GET /api/teaching-tasks?teacherId=&termId=
 */
export async function listTasksByTeacher(
  teacherId: number,
  termId?: number,
): Promise<TeachingTask[]> {
  const data = await http.get<unknown>('/teaching-tasks', {
    teacherId,
    ...(termId ? { termId } : {}),
  })
  return normalizeList<TeachingTask>(data)
}
